tests/test_update_ssh_keys.rs

```diff
@@ -0,0 +1,524 @@
+
+        TestContext {
+            ssh_dir: ssh_dir_path,
+        }
+    }
+
+    fn test_no_keys(ssh_dir: &str) {
+        let aks = open_authorized_keys(ssh_dir);
+
+        aks.write().expect("failed to update authorized keys directory");
+        assert!(format!("{}", aks.sync().unwrap_err().kind()).contains("no keys found"));
+
+        let authkeysdir: PathBuf = PathBuf::from(format!("{}/authorized_keys.d", ssh_dir));
+        assert!(authkeysdir.is_dir());
+
+        let authkeys: PathBuf = PathBuf::from(format!("{}/authorized_keys", ssh_dir));
+        assert!(!authkeys.exists());
+    }
+
+    fn test_first_run(ssh_dir: &str) {
+        // It is necessary to clean up before running the first run test,
+        // because old_authorized_keys will be only created when authorized_keys.d
+        // does not exist.
+        _ = fs::remove_dir_all(format!("{}/authorized_keys.d", ssh_dir));
+
+        let authkeys: PathBuf = PathBuf::from(format!("{}/authorized_keys", ssh_dir));
+        let mut authkeysfile =
+            File::options().create(true).append(true).write(true).open(authkeys.clone()).expect(format!("unable to create a file {}", authkeys.to_str().unwrap_or_default()).as_str());
+        for (_, text) in &*TEST_KEYS {
+            _ = authkeysfile.write_all(format!("{}\n", text).as_bytes());
+        }
+
+        // equivalent of running "update-ssh-keys" without args
+        let aks = open_authorized_keys(ssh_dir);
+
+        aks.write().expect("failed to update authorized keys directory");
+        aks.sync().expect("failed to update authorized keys");
+
+        let authkeys: PathBuf = PathBuf::from(format!("{}/authorized_keys.d/old_authorized_keys", ssh_dir));
+        assert!(authkeys.exists());
+        assert_has_keys(ssh_dir, ["valid1", "valid2", "valid3"].to_vec(), true);
+    }
+
+    #[test]
+    fn test_add_one_file() {
+        let ctx = setup_tests();
+
+        // "update-ssh-keys --add one valid1.pub"
+        _ = add_one_ssh_key(
+            ctx.ssh_dir.to_str().unwrap(), // ssh_dir
+            "valid1",                      // pubkeyname
+            "one",                         // testname
+            false,                         // is_force
+            true,                          // is_replace
+            ["valid1"].to_vec(),           // assert_keys
+        );
+    }
+
+    #[test]
+    fn test_replace_one() {
+        let ctx = setup_tests();
+
+        // "update-ssh-keys --add one valid1.pub"
+        _ = add_one_ssh_key(
+            ctx.ssh_dir.to_str().unwrap(), // ssh_dir
+            "valid1",                      // pubkeyname
+            "one",                         // testname
+            false,                         // is_force
+            true,                          // is_replace
+            ["valid1"].to_vec(),           // assert_keys
+        );
+
+        // "update-ssh-keys --add one valid2.pub"
+        _ = add_one_ssh_key(
```

Comment:
This supresses the error, please handle it

Reply:
Done